client/webserver/site/src/js/markets.js

```diff
@@ -692,7 +692,14 @@ export default class MarketsPage extends BasePage {
       if (!app.checkResponse(res)) return
       bttn.parentNode.textContent = 'cancelling'
       order.cancelling = true
-    })
+
+      // If "once" option is not supported by most browsers, we will need to
+      // unbind this function:
+      // Doc.unbind(page.cancelSubmit,'click', handler)
+
+      // What if the user clicks off the cancel form instead?  The cancelSubmit
+      // would bind another listener.
```

Comment:
I don't have an answer to this case though.  Seems like additional listeners will still be registered on each `showCancel` and they would not be removed unless the listener is called.  One solution would be to define a named function at a higher scope so `showCancel` could manually `removeEventListener` before binding a new one.

Reply:
I changed it to add the event handler once on market page construction.  That removes the need to manually remove the event listener after it is called or when the form is clicked away without submit.  I just provided the click data (order to cancel and row to modify on cancel) via the `cancelSubmit` object, which is the event target.